app/index.js

```diff
@@ -1,262 +1,640 @@
-        } else if (user.heartRateZone(hrm.heartRate) === 'peak') {
-            hrLabel.style.fill = 'fb-red'; //pink
-        } else if (user.heartRateZone(hrm.heartRate) === 'out-of-range') {
-            hrLabel.style.fill = 'fb-green'; //blue
+// Main view that shows the buttons in the flow
+const jsonFlow2 = document.getElementById("json-flow2"); // 2-answers' questions view
+const jsonFlow = document.getElementById("json-flow");   // 3-answers' questions view
+const jsonFlow4 = document.getElementById("json-flow4"); // 4-answers' questions view
+
+// Array containing all buttons
+let allButtons = [{
+    value: 10,
+    obj: comfy,
+    attribute: "startSurvey",
+}, {
+    value: 9,
+    obj: notComfy,
+    attribute: "startSurvey",
+},
+{
+    value: "flowBack",
+    obj: flowBack,
+    attribute: "flowControl",
+}, {
+    value: "flowStop",
+    obj: flowStop,
+    attribute: "flowControl",
+}, {
+    value: 9,
+    obj: centerButton3,
+    attribute: "question",
+}, {
+    value: 10,
+    obj: rightButton3,
+    attribute: "question",
+}, {
+    value: 11,
+    obj: leftButton3,
+    attribute: "question",
+}, {
+    value: 10,
+    obj: rightButton2,
+    attribute: "question",
+}, {
+    value: 11,
+    obj: leftButton2,
+    attribute: "question",
+}, {
+    value: 9,
+    obj: centerButton4,
+    attribute: "question",
+}, {
+    value: 10,
+    obj: rightButton4,
+    attribute: "question",
+}, {
+    value: 11,
+    obj: leftButton4,
+    attribute: "question",
+}, {
+    value: 12,
+    obj: bottomButton4,
+    attribute: "question",
+}, ];
+
+// Array containing all the views
+let viewsArray = [{
+    viewAttribute: "clockface",
+    viewObj: clockface,
+    viewQuestion: null
+},
+{
+    viewAttribute: "thankyou",
+    viewObj: thankyou,
+    viewQuestion: null
+},
+{
+    viewAttribute: "clockblock",
+    viewObj: clockblock,
+    viewQuestion: null
+},
+{
+    viewAttribute: "svg_stop_survey",
+    viewObj: svg_stop_survey,
+    viewQuestion: null
+}];
+
+for (let index = 0; index < questionsFlow.length; index++)
+{
+    viewsArray.push(jsonQuestionToView(questionsFlow, index));
+}
+
+let feedbackData;                        // Global variable gathering all data to feed back
+let nextView;                            // Temp for the next view to be shown
+let viewsStack = [];                     // Initialization of the stack
+viewsStack = initStack(viewsStack);
+
+/* -----------------------------------------------------------------------------------------------
+                                          MAIN PROCESS
+----------------------------------------------------------------------------------------------- */
+
+// Show clockface
+viewDisplay(viewsArray[0]);
+
+// Assign action on every button, taking into account the state of the stack and the button's attribute
+for (const button of allButtons)
+{
+    button.obj.addEventListener('mousedown', () => {
+        console.log('Button clicked: ' + button.attribute);
```

Comment:
Is button.attribute the best log? This just results in the logging of 'Button clicked question' instead of 'button clicked warmer' for example